### PR TITLE
Emit initial state for on_offline_message 

### PR DESCRIPTION
**Pull request by kiselev-dv**
Use inintial sate as default `on_offline_state` state.

### CODE REVIEW

**Comment by kiselev-dv** [comment]
test

**Comment by kiselev-dv** [comment]
@wzin that's the failed test:
```
22:56:02 RAN TEST: pep8
22:56:02 Got exit code 256
```
How to find what's wrong?

File: lg_common/scripts/uscs_service.py [hunk 1]
```diff
@@ -37,14 +37,17 @@ def set_url(param):
...
+                                 port, param[1:],
+                                 global_dependency_timeout)
         return url
 
     rospy.init_node(NODE_NAME, anonymous=False)
 
     director_topic = rospy.get_param('~director_topic', '/director/scene')
     message_topic = rospy.get_param('~message_topic', '/uscs/message')
-    connectivity_topic = rospy.get_param('~connectivity_topic', '/lg_offliner/offline')
+    offline_topic = rospy.get_param('~offline_topic', '/lg_offliner/offline')
```
Comment (wzin) [inline lg_common/scripts/uscs_service.py]: offline topic sounds better!

File: lg_common/scripts/uscs_service.py [hunk 2]
```diff
@@ -68,7 +71,7 @@ def set_url(param):
     )
 
     rospy.Subscriber(director_topic, GenericMessage, us.update_uscs_message)
-    rospy.Subscriber(connectivity_topic, Bool, us.handle_connectivity_message)
+    rospy.Subscriber(offline_topic, Bool, us.handle_offline_message)
```
Comment (wzin) [inline lg_common/scripts/uscs_service.py]: :+1:

File: lg_common/src/lg_common/uscs_service.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def __init__(self,
             rospy.loginfo("Enabling on_offline_state handling")
             self.on_offline_state = self._grab_scene(on_offline_state_scene_url)
         else:
-            rospy.loginfo("Disabling on_offline_state handling")
+            rospy.loginfo("Will use initial state for on_offline_state")
```
Comment (wzin) [inline lg_common/src/lg_common/uscs_service.py]: That's a good loginfo. I cant imagine a state where dont want to revert to default state when going offline.

File: lg_common/src/lg_common/uscs_service.py [hunk 4]
```diff
@@ -61,37 +61,38 @@ def __init__(self,
...
-        if not self.on_offline_state:
+        if self.state is None:
+            self.state = USCSMessageResponse()
+
+        if self.on_offline_state is None:
             """
             By default, when offline, go to initial state if there is no offline state
             explicitly specifyed.
             """
             self.on_offline_state = self.state
```
Comment (wzin) [inline lg_common/src/lg_common/uscs_service.py]: that's OK - we should handle offline state at all time. I still cant imagine a case where we dont want to do that.

File: lg_common/src/lg_common/uscs_service.py [hunk 5]
```diff
@@ -61,37 +61,38 @@ def __init__(self,
...
 
-        If data == True then emit on_online_state message
-        If data == False then emit on_offline_state message
+        If data == True then emit on_offline_state message
+        If data == False then emit on_online_state message
         """
+        rospy.loginfo("Incoming message: %s. Currently offline: %s" % (message, self.offline))
         if self.director_scene_publisher:
-            if message.data is True and self.on_online_state:
+            self.offline = message.data
```
Comment (wzin) [inline lg_common/src/lg_common/uscs_service.py]: This is going to overwrite the state and lead to a following bug:
- you are online
- you emit offline = False message
- you revert to default state

^ that's bad - you should listen on messages and set the state idempotently. So set the offline/online ivar only when it's different than the state.

**Comment by kiselev-dv** [comment]
Test

**Pull request merged**